File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import date

class DailyAttendanceStat(BaseModel):
    date: str
    present: int
    absent: int
    late: int

class ClassAttendanceStat(BaseModel):
    classId: str
    className: str
    attendancePercentage: float

class DashboardAnalytics(BaseModel):
    totalStudents: int
    presentToday: int
    absentToday: int
    attendancePercentage: float
    activeClasses: int
    trend: List[DailyAttendanceStat]

File: frontend/src/types/analytics.ts
// Analytics types aligned with backend schemas (app/schemas/analytics.py)

export interface DailyAttendanceStat {
  date: string;
  present: number;
  absent: number;
  late: number;
}

export interface ClassAttendanceStat {
  classId: string;
  className: string;
  attendancePercentage: number;
}

export interface DashboardAnalytics {
  totalStudents: number;
  presentToday: number;
  absentToday: number;
  attendancePercentage: number;
  activeClasses: number;
  trend: DailyAttendanceStat[];
}
